concat_yaml.py

```diff
@@ -0,0 +1,321 @@
+
+    if version not in spec_joint:
+        raise ValueError(
+            f'Cuda version {version} not provided by yaml from templates')
+
+    specs_module = []
+    prefix_module = []
+    for i in range(len(module_packages['cuda']['externals'])):
+        specs_module.append(module_packages['cuda']['externals'][i]['spec'])
+        prefix_module.append(module_packages['cuda']['externals'][i]['prefix'])
+
+    i = 0
+    found_cuda_version = False
+    for spec in specs_module:
+        if version in spec:
+            prefix = prefix_module[i]
+            found_cuda_version = True
+            break
+        i += 1
+
+    if not found_cuda_version:
+        raise ValueError(
+            f'Cuda version {version} not provided by spack-config module')
+
+    joint_packages['packages']['cuda']['externals'][0]['prefix'] = prefix
+
+    return joint_packages
+
+
+def rename_cray_mpich_to_mpich(packages):
+    '''
+    Rename cray-mpich from spack-config module
+    to mpich to be compatible with spack-c2sm
+    '''
+
+    print('Rename cray-mpich to mpich')
+    cray_mpich = packages['packages']['cray-mpich']
+
+    spec = cray_mpich['externals'][0]['spec']
+    spec = spec.replace('cray-', '')
+
+    cray_mpich['externals'][0]['spec'] = spec
+
+    packages['packages']['mpich'] = cray_mpich
+
+    packages['packages']['mpich']['buildable'] = False
+
+    packages['packages'].pop('cray-mpich')
+
+    return packages
+
+
+def allow_xml_to_be_built(packages):
+    print('Allow building of xml')
+    packages['packages']['libxml2']['buildable'] = True
+    return packages
+
+
+# SPACK COMMANDS
+
+
+def spack_external_find(machine, packages_file):
+    '''
+    run spack external find and write
+    packages.yaml to current workingdir
+    '''
+
+    print(f'Find externals on {machine}')
+
+    os.environ["SPACK_USER_CONFIG_PATH"] = os.getcwd()
+
+    if os.path.exists(packages_file): os.remove(packages_file)
+
+    command = [
+        "./config.py",
+        "-i",
+        ".",
+        "-u",
+        "OFF",
+        "-m",
+        machine,
+        "--no_yaml_copy",
+    ]
+    subprocess.run(command, check=True)
+    command = [
+        'bash', '-c', "source spack/share/spack/setup-env.sh && \
+               spack external find --not-buildable --scope=user"
+    ]
+    subprocess.run(command, check=True)
+
+    os.environ.pop("SPACK_USER_CONFIG_PATH")
+
+
+# MERGE OF INDIVIDUAL YAML-FILES
+
+
+def remove_duplicate_compilers(c2sm, cscs, keys):
+    c2sm_specs = specs_from_list_with_keys(c2sm, keys[0], keys[1])
+    cscs_specs = specs_from_list_with_keys(cscs, keys[0], keys[1])
+
+    duplicates = (c2sm_specs & cscs_specs)
+    for dupl in duplicates:
+        cscs_specs.remove(dupl)
+
+    c2sm = [item for item in c2sm if item[keys[0]][keys[1]] in c2sm_specs]
+    cscs = [item for item in cscs if item[keys[0]][keys[1]] in cscs_specs]
+
+    return c2sm + cscs
+
+
+def remove_duplicate_packages(c2sm, cscs, external):
```

Comment:
The name of this function is misleading, it doesn't remove anything. It returns something. Please find a better name.
Can't it be simplified to this?
```
def name_pending(primary, secondary, tertiary):
    tertiary.update(secondary)
    tertiary.update(primary)
    return tertiary
```

Reply:
AH I just discoverd that this function is not used at all!